contracts/pcv/curve/StableSwapOperatorV1.sol

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "../PCVDeposit.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+// Curve metapool
+interface IStableSwap {
+    function coins(uint256 arg0) external view returns (address);
+    function balances(uint256 arg0) external view returns (uint256);
+    function add_liquidity(uint256[2] memory amounts, uint256 min_mint_amount) external;
+    function remove_liquidity(uint256 _amount, uint256[2] memory min_amounts) external;
+    function remove_liquidity_one_coin(uint256 _token_amount, int128 i, uint256 min_amount) external;
+    function get_virtual_price() external view returns (uint256);
+    function calc_withdraw_one_coin(uint256 _token_amount, int128 i) external view returns (uint256);
+}
+interface I3Pool {
+    function add_liquidity(uint256[3] memory amounts, uint256 min_mint_amount) external;
+}
+
+/// @title StableSwapOperator: implementation for a Curve Metapool market-maker.
+/// This is version 1, it only allows simple deposits and does not require the
+/// Minter role. FEI has to be minted on this contract before calling deposit().
+/// There are no reweight mechanisms.
+/// All accounting and withdrawals are denominated in DAI. Should the contract
+/// hold USDC or USDT, these can also be deposited.
+/// @author eswak
+contract StableSwapOperatorV1 is PCVDeposit {
+    using SafeERC20 for ERC20;
+
+    // ------------------ Properties -------------------------------------------
+
+    uint256 public depositMaxSlippageBasisPoints;
+    uint256 public constant BASIS_POINTS_GRANULARITY = 10_000;
+
+    /// @notice The StableSwap pool to deposit in
+    address public pool;
+
+    // ------------------ Private properties -----------------------------------
+
+    /// some fixed variables to interact with the pool
+    uint8 private _feiIndex; // index of FEI in the pool (0 or 1)
+    uint8 private _3crvIndex; // index of 3crv in the pool (0 or 1)
+    address private _3crv; // address of the 3crv token
+    address private _3pool; // address of the 3pool
+    address private _dai; // address of the DAI token
+    address private _usdc; // address of the USDC token
+    address private _usdt; // address of the USDT token
```

Comment:
can make immutable to save gas. Immutable variables are compiled inline to save a SLOAD call

Reply:
done 👍 

Had to introduce a temporary variable for _3crvIndex because immutable variables cannot be read during contract creation